File: core/src/net/bossdragon/component/Soldier.java
package net.bossdragon.component;

import com.artemis.PooledComponent;

/**
 *
 */
public class Soldier extends PooledComponent {
    public float lyingCooldown = 0f;

    @Override
    protected void reset() {
        lyingCooldown = 0f;
    }

    public boolean isLying() {
        return lyingCooldown > 0f;
    }
}

File: core/src/net/bossdragon/component/Soldier.kt
package net.bossdragon.component

import com.artemis.PooledComponent

class Soldier : PooledComponent() {
    override fun reset() {
        lyingCooldown = 0f
    }

    var lyingCooldown: Float = 0f

    val isLying: Boolean
        get() = lyingCooldown > 0f
}
